app/src/main/java/com/sample/appconnectsample/ImageCaptureActivity.java

```diff
@@ -0,0 +1,193 @@
+        takePictureButton = (Button)findViewById(R.id.takePictureButton);
+
+        // Get the ID of the subject to save images for
+        Intent intent = getIntent();
+        subjectID = intent.getLongExtra(SUBJECT_ID_EXTRA, 0);
+
+        if (!getApplicationContext().getPackageManager().hasSystemFeature(PackageManager.FEATURE_CAMERA_ANY))
+            takePictureButton.setEnabled(false);
+
+        submitButton.setEnabled(false);
+
+    }
+
+    public void doSelectImageButton(View source) {
+        dispatchSelectImageIntent();
+    }
+
+    public void doTakePictureButton(View source) {
+        dispatchTakePictureIntent();
+    }
+
+    private static final int IMAGE_CAPTURE_INTENT = 1;
+    private static final int SELECT_IMAGE_INTENT = 2;
+
+    private void dispatchTakePictureIntent() {
+        Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        if (takePictureIntent.resolveActivity(getPackageManager()) != null) {
+            startActivityForResult(takePictureIntent, IMAGE_CAPTURE_INTENT);
+        }
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == IMAGE_CAPTURE_INTENT && resultCode == RESULT_OK) {
+            Bundle extras = data.getExtras();
+            currentImage = (Bitmap) extras.get("data");
+            currentImageView.setImageBitmap(currentImage);
+            submitButton.setEnabled(true);
+        }
+        else if (requestCode == SELECT_IMAGE_INTENT && resultCode == RESULT_OK) {
+            Uri selectedImage = data.getData();
+            try {
+                InputStream imageStream = getContentResolver().openInputStream(selectedImage);
+                currentImage = BitmapFactory.decodeStream(imageStream);
+                currentImageView.setImageBitmap(currentImage);
+                submitButton.setEnabled(true);
+            } catch (FileNotFoundException exception) { }
+        }
+    }
+
+    private void dispatchSelectImageIntent() {
+        Intent photoPickerIntent = new Intent(Intent.ACTION_PICK);
+        photoPickerIntent.setType("image/*");
+        startActivityForResult(photoPickerIntent, SELECT_IMAGE_INTENT);
+    }
+
+    public void doSubmitButton(View source) {
+        submitButton.setEnabled(false);
+        new CollectAndSubmitTask().execute(currentImage);
+    }
+
+    /**
+     * An asynchronous task that collects the responses and submits the form.
+     */
+    private class CollectAndSubmitTask extends AsyncTask<Bitmap,Void,Void> {
+
+        private ProgressDialog progressDialog;
+        private String message;
+        private Exception exception;
+
+        @Override
+        protected void onPreExecute() {
+            progressDialog = new ProgressDialog(ImageCaptureActivity.this, ProgressDialog.STYLE_SPINNER);
+            progressDialog.setMessage(getString(R.string.image_capture_saving_message));
```

Comment:
Nitpick: I think the resource ID can be passed directly to `setMessage` without needing `getString()`